pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -326,6 +327,89 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
         emit SwapEnabledSet(swapEnabled);
     }
 
+    /**
+     * @dev Removes a token from the Pool's composition, withdrawing all funds from the Vault and sending them to
+     * `recipient`, and adjusting the weights of all other tokens.
+     *
+     * Tokens can only be removed if the Pool has more than 2 tokens, as it can never have fewer than 2. Token removal
+     * is also forbidden during a weight change, or if one is scheduled to happen in the future.
+     *
+     * The caller may aditionally pass a non-zero `burnAmount` to have some of their BPT be burned, which might be
+     * useful in some scenarios to account for the fact that the Pool now has fewer tokens.
+     */
+    function removeToken(
+        IERC20 token,
+        address recipient,
+        uint256 burnAmount
+    ) external authenticate nonReentrant whenNotPaused {
+        // Exit the pool, returning the full balance of the token to the recipient
+        (IERC20[] memory tokens, uint256[] memory unscaledBalances, ) = getVault().getPoolTokens(getPoolId());
+        _require(tokens.length > 2, Errors.MIN_TOKENS);
+
+        // Tokens cannot be removed during or before a weight change to reduce complexity of weight interactions
+        _ensureNoWeightChange();
+
+        // Reverts if token does not exist in pool.
+        uint256 tokenIndex = _tokenAddressToIndex(tokens, token);
+        uint256 tokenBalance = unscaledBalances[tokenIndex];
```

Comment:
```suggestion
    ) external authenticate nonReentrant whenNotPaused returns (uint256 tokenAmountOut) {
        // Exit the pool, returning the full balance of the token to the recipient
        (IERC20[] memory tokens, uint256[] memory unscaledBalances, ) = getVault().getPoolTokens(getPoolId());
        _require(tokens.length > 2, Errors.MIN_TOKENS);

        // Tokens cannot be removed during or before a weight change to reduce complexity of weight interactions
        _ensureNoWeightChange();

        // Reverts if token does not exist in pool.
        uint256 tokenIndex = _tokenAddressToIndex(tokens, token);
        tokenAmountOut = unscaledBalances[tokenIndex];
```

This will make it easier for Pool managers to know how many tokens they receive (as opposed to before+after balance checks)

Reply:
Great suggestion, yes. I'll add it in a slightly different way as I don't like output variables :upside_down_face: